Add contributing documentation

rmccue opened the pull request:
Hoping this unblocks the process from relying on me and/or Joe.

mikeselander commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,34 @@
+
+Any changes which cause existing, working production code to fail should trigger a new major release. Only bugfixes or making rules more lenient should be in minor releases.
+
+When publishing major releases, these need to be published in a two-step process. First, publish the standards, then bump the defaults after some time. This gives projects time to assess the changes and migrate at their own pace.
```
Could you give an idea of how much time you'd expect for this period? Perhaps two weeks would be sufficient?

mikeselander commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,34 @@
+* Push the new tag: `git push --tags`
+	* This triggers Packagist to release the new version
+* Run `./publish.sh` to push the standards for hm-linter
+	* This will ask if you want to bump the latest version to the new version. Only do this for minor releases.
```
We should probably also add writing up release notes for our `dev` P2 and adding changelog notes on GitHub here.

rmccue commented:
@mikeselander Addressed your feedback here. For the time between release and default bump for major releases, generally 2+ sprints: assume you release after a sprint has already been scoped, so earliest the work can happen is in the next sprint. Even bigger changes (say, where you're trying to shift existing behaviour by making rules stricter) may need longer as the changes to the codebase may be larger.

rmccue commented:
Unsure why tests are failing here, is `master` broken?

mikeselander commented:
Looks like it, I'm going to take a look this morning. I wouldn't have expected your PR to be failing though 🤔
